.github/workflows/ci.yml

```
@@ -104,6 +110,9 @@ jobs:
         run: |
           cargo clippy --target=aarch64-apple-darwin -- --deny warnings
           cargo clippy --target=x86_64-apple-darwin -- --deny warnings
+      - uses: cachix/install-nix-action@v31
+      - name: nix build
+        run: nix build
```

You can select the build targets with

```suggestion
        run: |
          nix build --system arch64-darwin
          nix build --system x86_64-darwin
```

You could do the same for Linux. Though, I'm not sure if building succeeds on the github workers.

---

> Though, I'm not sure if building succeeds on the github workers.

It better! That's kind of the point of nix!

---

I would usually use the `nix build .#packages.aarch64-darwin.espanso` syntax (as reflected by `nix flake show`).

I'm happy to add the macos-intel nix target to this, but it's looking to be formally deprecated in nixpkgs in the near future: https://github.com/NixOS/nixpkgs/pull/415566